nano/rpc_test/rpc.cpp

```diff
@@ -2999,6 +2999,8 @@ TEST (rpc, work_cancel)
 		ASSERT_TIMELY (10s, response1.status != 0);
 		ASSERT_EQ (200, response1.status);
 		ASSERT_NO_ERROR (ec);
+		std::string success (response1.json.get<std::string> ("success"));
```

Comment:
Before doing `json.get...` you should check that "success` is there with e.g. `json.count ("success")`, otherwise it will throw an exception if not found (instead of properly failing the test). An alternative is using `get_optional` and checking that the return is initialized.

Reply:
Hmm.. do exceptions not fail the test properly? Sorry, I'm a java dev and don't work with C languages or frameworks very often.

None of the other tests appear to check the presence of an attribute, unless I'm reading them wrong. I had just copy-pasted the check from bootstrap_any.

Reply:
We haven't enforced it in the past so no worries.